app/controllers/agent/business/BusinessNameController.scala

```diff
@@ -52,19 +53,34 @@ class BusinessNameController @Inject()(val authService: AuthService, subscriptio
       ))
   }
 
+  private def redirectLocation( currentAnswer: BusinessNameModel, isEditMode: Boolean)(implicit request: Request[AnyContent]): Future[Result] = {
+    for {
+      businessName <- subscriptionDetailsService.fetchBusinessName()
+      incomeSources <- subscriptionDetailsService.fetchIncomeSource
+    } yield {
+      (businessName, incomeSources) match {
+        case (_,None) =>
+          Redirect(controllers.agent.routes.IncomeSourceController.show())
+        case (_,_) if isEditMode =>
+          Redirect(controllers.agent.routes.CheckYourAnswersController.show())
+        case (_, Some(IncomeSourceModel(true, true, _))) =>
+          Redirect(controllers.agent.business.routes.BusinessAccountingMethodController.show(isEditMode))
```

Comment:
```suggestion
          Redirect(controllers.agent.business.routes.BusinessAccountingMethodController.show())
```
isEditMode at this point can only ever be false, which is the default value for the show, can be removed